### ISSUE TITLE
JSON schema generated for date field is invalid upon calling `validate`

### ISSUE THREAD

**Issue by syastrov** [seandstewart/typical#21]
It seems like typical doesn't convert the StringFormat.DATE enum field into a primitive (e.g. str), which fastjsonschema doesn't like.

```python
>>> import typic
>>> from datetime import date
>>> from dataclasses import dataclass
>>> @typic.al
... @dataclass
... class Foo:
...   foo: date
...
>>> typic.schema(Foo)
ObjectSchemaField(title='Foo', description='Foo(foo: datetime.date)', properties={'foo': StrSchemaField(format=<StringFormat.DATE: 'date'>)}, additionalProperties=False, required=('foo',))
>>> typic.schema(Foo).validate({})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../venv/lib/python3.7/site-packages/typic/schema/field.py", line 194, in validate
    return self.validator(obj)
  File ".../venv/lib/python3.7/site-packages/typic/util.py", line 226, in __get__
    cache[attrname] = self.func(instance)
  File ".../venv/lib/python3.7/site-packages/typic/schema/field.py", line 189, in validator
    return fastjsonschema.compile(self.asdict())
  File ".../venv/lib/python3.7/site-packages/fastjsonschema/__init__.py", line 167, in compile
    exec(code_generator.func_code, global_state)
  File "<string>", line 5
    raise JsonSchemaException("data must be object", value=data, name="data", definition={'type': 'object', 'title': 'Foo', 'description': 'Foo(foo: datetime.date)', 'properties': {'foo': {'type': 'string', 'format': <StringFormat.DATE: 'date'>}}, 'additionalProperties': False, 'required': ['foo'], 'definitions': {}}, rule='type')
                                                                                                                                                                                                                         ^
SyntaxError: invalid syntax
```

**Comment by seandstewart** [comment]
Hey @syastrov, this should be resolved with the latest release (v2.0.0b7)

```python
>>> import typic
>>> import datetime
>>> @typic.klass
... class Bar:
...     bar: datetime.datetime
...     
>>> Bar.schema().validate({})
Traceback (most recent call last):
 ...
fastjsonschema.exceptions.JsonSchemaException: data must contain ['bar'] properties
>>> Bar.schema().primitive()
{'type': 'object', 'title': 'Bar', 'description': 'Bar(bar: datetime.datetime)', 'properties': {'bar': {'type': 'string', 'format': 'date-time'}}, 'additionalProperties': False, 'required': ['bar'], 'definitions': {}}
```

**Comment by syastrov** [comment]
@seandstewart This seems to be broken again in 2.0.27.

**Comment by seandstewart** [comment]
@syastrov sorry for the regression here! I've just released a patch in `v2.0.28`.

**Comment by syastrov** [comment]
Thanks! and what a fast fix 👍
